hudi-client/hudi-client-common/src/main/java/org/apache/hudi/schema/SchemaProviderInterface.java

```diff
@@ -25,12 +25,13 @@
 /**
  * Class to provide schema for reading data and also writing into a Hoodie table.
  */
-public abstract class SchemaProvider implements Serializable {
+public interface SchemaProviderInterface extends Serializable {
 
-  public abstract Schema getSourceSchema();
+  Schema getSourceSchema();
```

Comment:
shouldn't we add 
@PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
also for tagetSchema method as well.